apps/omg/lib/omg/state/transaction/payment.ex

```diff
@@ -128,15 +128,17 @@ defmodule OMG.State.Transaction.Payment do
   defp reconstruct_metadata(_), do: {:error, :malformed_metadata}
 
   defp parse_inputs(inputs_rlp) do
-    {:ok, Enum.map(inputs_rlp, &parse_input!/1)}
+    with true <- Enum.count(inputs_rlp) <= @max_inputs || {:error, :too_many_inputs},
+         do: {:ok, Enum.map(inputs_rlp, &parse_input!/1)}
   rescue
     _ -> {:error, :malformed_inputs}
```

Comment:
this is actually not your code but I've spotted it now. It might be better to have a stricter validation here, this rescue will catch everything and quite possibly catch all underlying issues. or am I reading it wrong?

Reply:
yeah, I was pondering this too, I think the original intention was to simplify the inner handling (that all inputs must satisfy a bunch of criteria) and always put any kind of invalidity into the `malformed_inputs` bag.

But maybe refactoring this to a bunch of `reduce`s is better. As things evolve the `rescue` does start to sound a bit scary.

Reply:
I put that under https://github.com/omisego/elixir-omg/issues/1245 to handle separately and not bloat this one further